### ISSUE TITLE
[CODE] Using some Python coding best practices for open source repositories

### ISSUE THREAD

**Issue by fabianegli** [bigbio/sdrf-pipelines#105]
I propose that tis project adopts some of the standard Python development tools to make contributions easier and reduce some of the burden of new contributors.

* `black` and `black-disable-checker` for code formatting
* `isort` to systematically sorting the imports
* `pylint` to spot and fix mistakes, errors and code smells

Further tools that are not yet very useful here but might become handy in the future

* `blacken-docs` for documentation
* `mypy` if type annotations become a thing in this package.

**Comment by ypriverol** [comment]
Agreed. Do you know how we can enforce that by code or linting etc. Another question, do you know how to validate that with IDE as Pycharm?

**Comment by fabianegli** [comment]
For the branch protection an admin needs to go to Setttings -> Branches and then click the "Add rule" in the "Branch protection rules" section, type the name of the branch to be protected (master) and then check at least the top two protections "Require a pull request before merging" and "Require status checks to pass before merging". That should do it.

**Comment by ypriverol** [comment]
I will protect the master.

**Comment by ypriverol** [comment]
> For the branch protection an admin needs to go to Setttings -> Branches and then click the "Add rule" in the "Branch protection rules" section, type the name of the branch to be protected (master) and then check at least the top two protections "Require a pull request before merging" and "Require status checks to pass before merging". That should do it.

**DONE.**

**Comment by fabianegli** [comment]
Because the blacken-docs is currently out of scope for this repo because there are no docs I think that the current CI checks and the ones in #205 will suffice for this repository.

**Comment by fabianegli** [comment]
As an alternative to blacken-docs it could be worthwhile to add docformatter, but that would require changes to the current docstring style in some files. Docstrings do not currently follow a consistent style or formatting.
